sweep/walletsweep.go

```diff
@@ -66,3 +70,207 @@ func DetermineFeePerKw(feeEstimator lnwallet.FeeEstimator,
 		return feePerKw, nil
 	}
 }
+
+// UtxoSource is an interface that allows a caller to access a source of UTXOs
+// to use when crafting sweep transactions.
+type UtxoSource interface {
+	// ListUnspentWitness returns all UTXOs from the source that have
+	// between minConfs and maxConfs number of confirmations.
+	ListUnspentWitness(minConfs, maxConfs int32) ([]*lnwallet.Utxo, error)
+
+	// FetchInputInfo returns the matching output for an outpoint. If the
+	// outpoint doesn't belong to this UTXO source, then an error should be
+	// returned.
+	FetchInputInfo(*wire.OutPoint) (*wire.TxOut, error)
+}
+
+// CoinSelectionLocker is an interface that allows the caller to perform an
+// operation, which is synchronized with all coin selection attempts. This can
+// be used when an operation requires that all coin selection operations cease
+// forward progress. Think of this as an exclusive lock on coin selection
+// operations.
+type CoinSelectionLocker interface {
+	// WithCoinSelectLock will execute the passed function closure in a
+	// synchronized manner preventing any coin selection operations from
+	// proceeding while the closure if executing. This can be seen as the
+	// ability to execute a function closure under an exclusive coin
+	// selection lock.
+	WithCoinSelectLock(func() error) error
+}
+
+// OutpointLocker allows a caller to lock/unlock an outpoint. When locked, the
+// outpoints should t be used for any sort of channel funding of coin
+// selection. Locked outpoints are not expect to be persisted between restarts.
+type OutpointLocker interface {
+	// LockOutpoint locks a target outpoint, rendering it unusable for coin
+	// selection.
+	LockOutpoint(o wire.OutPoint)
+
+	// UnlockOutpoint unlocks a target outpoint, allowing it to be used for
+	// coin selection once again.
+	UnlockOutpoint(o wire.OutPoint)
```

Comment:
nit: parameter name not needed

Reply:
Many other interfaces in the codebase name the interface params.